azavea/echo-locator
# Bootstrap Django API

## Overview

This PR:
- adds `django` and `database` services for local dev as API and DB
- adds/updates `./scripts/bootstrap`, `./scripts/setup`, `./scripts/update`, `./scripts/manage`, `./scripts/server`, `./scripts/dbshell`
- adds a simple `/api/` API route and returns a message that shows the API works as expected
- adds a small test on the frontend to allow testing the communication with local dev API

### Notes

1. I made some notes as single comments in the files changed below. Feel free to discuss!
2. I did not add in Vagrant in this solution, and would like to propose to start from just using Docker. If we find it easier to add Vagrant as a layer in between in the future, I'd be happy to take that issue as well.

## Testing Instructions

 * Make sure your local docker app is running
 * At the root of the directory, run `./scripts/setup`, which should bootstrap the app, install deps, build containers, and run migrations
 * Run `./scripts/server`, which should spin up a local DB, frontend, and Django API server
 * After the Django API is up and running, make sure the following is successful
```
curl --location --request GET 'http://localhost:8081/api'
```
* Go to http://localhost:9966, open your network tabs in the dev tools, and filter to the `/api/` call. Make sure the request is also successful
* At the root of the directory, run `./scripts/manage migrate` and make sure it runs migrations, although there is no migration to run now, which is fine
~At the root of the directory, run `./scripts/dbshell` and make sure it takes you to a local `psql` shell and you could do normal SQL stuffs there~


Resolves #414
